lms/djangoapps/discussion/tests/test_tasks.py

```diff
@@ -0,0 +1,151 @@
+"""
+Tests the execution of forum notification tasks.
+"""
+from contextlib import contextmanager
+from datetime import datetime, timedelta
+import json
+import math
+from urlparse import urljoin
+
+import ddt
+from django.conf import settings
+from django.contrib.sites.models import Site
+import mock
+
+from django_comment_common.models import ForumsConfig
+from django_comment_common.signals import comment_created
+from edx_ace.recipient import Recipient
+from lms.djangoapps.discussion.config.waffle import waffle, FORUM_RESPONSE_NOTIFICATIONS
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+from openedx.core.djangoapps.schedules.template_context import get_base_template_context
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+
+@contextmanager
+def mock_the_things():
```

Comment:
Did you just pull all the mocks out of the test function because of the weird interactions with ddt?

Reply:
Well, that and having some problems with pep8 and indentation.